Vic2ToHoI4/Source/HOI4World/HoI4FocusTree.cpp

```diff
@@ -1932,8 +2032,223 @@ void HoI4FocusTree::addGPWarBranch(shared_ptr<HoI4::Country> Home,
 }
 
 
+std::set<std::string> HoI4FocusTree::addConquerBranch(
+	std::shared_ptr<HoI4::Country> theCountry,
+	int& numWarsWithNeighbors,
+	const std::set<std::string>& majorIdeologies,
+	 HoI4::Localisation& hoi4Localisations)
+{
+	std::string tag = theCountry->getTag();
+	std::vector<std::pair<std::string, int>> conquerQueue;
+	std::set<std::string> conquerTags;
+
+	const auto& conquerStrategies = theCountry->getConquerStrategies();
+	if (conquerStrategies.empty())
+	{
+		return conquerTags;
+	}
+	const auto& maxConquerValueItr = std::max_element(conquerStrategies.begin(), conquerStrategies.end(), [](HoI4::AIStrategy a, HoI4::AIStrategy b) {
+		return a.getValue() < b.getValue();
+	});
+	int maxConquerValue = maxConquerValueItr->getValue();
+
+	for (const auto& strategy: conquerStrategies)
+	{
+		if (theCountry->isEligibleEnemy(strategy.getID()))
+		{
+			conquerQueue.push_back(make_pair(strategy.getID(), strategy.getValue()));
+		}
+	}
+	std::sort(conquerQueue.begin(), conquerQueue.end(), [](std::pair<std::string, int>a, std::pair<std::string, int>b) {
+		return a.second > b.second;
+	});
+
+	if (!conquerQueue.empty())
+	{
+		for (const auto& target: conquerQueue)
+		{
+			int aiChance = 10 * target.second / maxConquerValue;
```

Comment:
> This is vulnerable to integer truncation, and will come out to zero in most cases.

It is meant to be a modifier to the 10 base `ai_chance`. `target.second` is `conquer_strategy` value, while `maxConquerValue` is the value for the top country among those that are to be conquered. I figured, when `target.second` is no more than 20 times smaller than the `maxConquerValue`, the truncation would round `aiChance` to 1.

Reply:
Oops, I was reading the order of operations wrong. Still, you may wish to std::clamp it to ensure its in the range of 1-10 if that's your intent.

Reply:
> Oops, I was reading the order of operations wrong. Still, you may wish to std::clamp it to ensure its in the range of 1-10 if that's your intent.

After taking another look, I actually can't have it clamped 1-10 because I'm breaking the loop if it's <1. Is there something else I can/need to do instead of std::clamp?

Reply:
No, I think it's fine in that case. It should naturally fall in the 0-10 range.